app/src/org/commcare/android/util/FormUploadUtil.java

```diff
@@ -62,262 +70,308 @@ public static Cipher getDecryptCipher(SecretKeySpec key) {
-                    if(!validateSubmissionFile(f, FormUploadUtil.getDecryptCipher(key))) {
-                        return RECORD_FAILURE;
+                if (key != null) {
+                    if (!validateSubmissionFile(f)) {
+                        return false;
                     }
-                    fb = new EncryptedFileBody(f, FormUploadUtil.getDecryptCipher(key), "text/xml");
-                }
-                else{
+                    fb = new EncryptedFileBody(f, FormUploadUtil.getDecryptCipher(key),
+                            "text/xml");
+                } else {
                     fb = new FileBody(f, "text/xml");
                 }
-                
                 entity.addPart("xml_submission_file", fb);
-                
-                //fb = new FileBody(f, "text/xml");
-                //Don't know if we can ask for the content length on the input stream, so skip it.
-//                if (fb.getContentLength() <= MAX_BYTES) {
-//                    Log.i(t, "added xml file " + f.getName());
-//                } else {
-//                    Log.i(t, "file " + f.getName() + " is too big");
-//                }
             } else if (f.getName().endsWith(".jpg")) {
                 fb = new FileBody(f, "image/jpeg");
                 if (fb.getContentLength() <= MAX_BYTES) {
                     entity.addPart(f.getName(), fb);
-                    Log.i(t, "added image file " + f.getName());
+                    Log.i(TAG, "added image file " + f.getName());
                 } else {
-                    Log.i(t, "file " + f.getName() + " is too big");
+                    Log.i(TAG, "file " + f.getName() + " is too big");
                 }
             } else if (f.getName().endsWith(".3gpp")) {
                 fb = new FileBody(f, "audio/3gpp");
                 if (fb.getContentLength() <= MAX_BYTES) {
                     entity.addPart(f.getName(), fb);
-                    Log.i(t, "added audio file " + f.getName());
+                    Log.i(TAG, "added audio file " + f.getName());
                 } else {
-                    Log.i(t, "file " + f.getName() + " is too big");
+                    Log.i(TAG, "file " + f.getName() + " is too big");
                 }
             } else if (f.getName().endsWith(".3gp")) {
                 fb = new FileBody(f, "video/3gpp");
                 if (fb.getContentLength() <= MAX_BYTES) {
                     entity.addPart(f.getName(), fb);
-                    Log.i(t, "added video file " + f.getName());
+                    Log.i(TAG, "added video file " + f.getName());
                 } else {
-                    Log.i(t, "file " + f.getName() + " is too big");
+                    Log.i(TAG, "file " + f.getName() + " is too big");
+                }
+            } else if (supported || isAudioVisualMimeType(f)) {
+                fb = new FileBody(f, "application/octet-stream");
+                if (fb.getContentLength() <= MAX_BYTES) {
```

Comment:
The actual bug fix: Implemented and made call to isAudioVisualMimeType. This ensures that queued up files with the correct mimetype get added to the multipart entity being submitted to HQ.